src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -227,27 +233,43 @@ def add_imaging_plane(nwbfile: NWBFile, metadata: dict, imaging_plane_index: int
         An previously defined -in memory- NWBFile.
     metadata : dict
         The metadata in the nwb conversion tools format.
-    imaging_plane_index : int, default: 0
-        The metadata in the nwb conversion tools format is a list of the different imaging planes to add.
-        Specify which element of the list with this parameter.
-
+    imaging_plane_name: str
+        The name of the imaging plane to be added.
     Returns
     -------
     NWBFile
         The nwbfile passed as an input with the imaging plane added.
     """
+    if imaging_plane_index is not None:
+        warn(
+            message="Keyword argument 'imaging_plane_index' is deprecated and will be removed on or after Dec 1st, 2023. "
+            "Use 'imaging_plane_name' to specify which imaging plane to add by its name.",
+            category=DeprecationWarning,
+        )
+        imaging_plane_name = metadata["Ophys"]["ImagingPlane"][imaging_plane_index]["name"]
 
     # Set the defaults and required infrastructure
     metadata_copy = deepcopy(metadata)
     default_metadata = get_default_ophys_metadata()
     metadata_copy = dict_deep_update(default_metadata, metadata_copy, append_list=False)
     add_devices(nwbfile=nwbfile, metadata=metadata_copy)
 
-    imaging_plane_metadata = metadata_copy["Ophys"]["ImagingPlane"][imaging_plane_index]
-    imaging_plane_name = imaging_plane_metadata["name"]
+    default_imaging_plane_name = metadata_copy["Ophys"]["ImagingPlane"][0]["name"]
```

Comment:
@h-mayorquin Now that I'm looking at it, the logic that I wanted to use here is when you don't specify the name of the imaging plane, it will look up the "default" from the metadata. But it's not actually the default name "ImagingPlane" but after it has been updated in `metadata_copy`.
Does that make sense?

Reply:
Let me think.
If we leave `metadata_copy` and the user passess some imaging plane metadata (but they don't specify`imaging_plane_name`) then this will choose the first name of the first element in their `metadata["Ophys"]["ImagingPlane"]` list. Correct?

Why would this be a good default and better than the value from `default_metadata`? How are you thinking about it?

Reply:
@h-mayorquin I'm not sure my reasoning was correct; but if we are comparing to the default_metadata then we would expose `imaging_plane_name` to `add_photon_series` and `add_segmentation`, because the user would have to define the name of the imaging plane they want to use in 1) get_metadata() and in 2) add_to_nwbfile(imaging_plane_name=""). Comparing to metadata_copy would silently choose it for them, but maybe that is not the good default.

Reply:
I am not sure I understand case 1.

Let me say something though, I think we don't need to expose the `imaging_plane_name` in `add_photon_series`. The metadata for the photon series already has the name of the plane on it. We should just swap these two lines:

https://github.com/catalystneuro/neuroconv/blob/0015f1a3b668a7dedef7ff391e4486bfe19e8bd1/src/neuroconv/tools/roiextractors/roiextractors.py#L364-L365

As for `add_segmentation` imaging_plane is an also a keyword argument of Segmentation:
https://pynwb.readthedocs.io/en/stable/pynwb.ophys.html#pynwb.ophys.PlaneSegmentation

So probably just having a default name in the metadaata for `get_default_segmentation_metadata` should work?

Reply:
Right, I think that could work! Thanks!

Reply:
Fixed the tests, ready for the final review @h-mayorquin